src/controlers/cart.controller.ts

```diff
@@ -0,0 +1,175 @@
+import Article from '@models/article.model';
+import Cart from '@models/cart.model';
+import { CartArticle } from '@models/cartArticle.model';
+import Color from '@models/color.model';
+import Size from '@models/size.model';
+import { Request, Response } from 'express';
+
+export const createCart = async (req: Request, res: Response) => {
+  try {
+    // const { userId } = req.body.userId;
+    const userId = req.session.user.id;
+    const existingCart = await Cart.findOne({
+      where: {
+        userId,
+      },
+    });
+
+    if (existingCart) {
+      return res.status(200).json({ message: 'Cart already exists' });
+    }
+
+    const cart = await Cart.create({
+      userId,
+    });
+    return res.status(200).json(cart);
+  } catch (error) {
+    return res.json(error);
+  }
+};
+
+export const getCart = async (req: Request, res: Response) => {
+  try {
+    const userId = req.session.user.id;
+    const cart = await Cart.findOne({
+      where: {
+        userId,
+      },
+      include: [
+        {
+          model: Article,
+          include: [{ model: Color }, { model: Size }],
+        },
+      ],
+    });
+
+    if (!cart) {
+      return res.status(404).json({ message: 'Cart not found' });
+    }
+
+    return res.status(200).json(cart);
+  } catch (error) {
+    console.error('Error retrieving cart:', error);
+    res.status(500).json({ error: 'Internal server error' });
+  }
+};
+
+export const addArticleToCart = async (req: Request, res: Response) => {
+  try {
+    const articleId = req.body.articleId;
+    const cartId = req.body.cartId;
+
+    const cart = await Cart.findOne({
+      where: {
+        id: cartId,
+      },
+      include: [{ model: Article }],
+    });
+
+    if (!cart) {
+      return res.status(404).json({ message: 'Cart not found' });
+    }
+
+    const articleExists = cart.articles.some(
+      (article) => article.id === articleId
+    );
+
+    if (articleExists) {
+      const cartArticle = await CartArticle.findOne({
+        where: {
+          cartId,
+          articleId,
+        },
+      });
+
+      await cartArticle.update({
+        quantity: cartArticle.quantity + 1,
+      });
+
+      const updatedCart = await Cart.findOne({
+        where: {
+          id: cartId,
+        },
+        include: [
+          { model: Article, include: [{ model: Color }, { model: Size }] },
+        ],
+      });
+      return res.status(200).json(updatedCart);
+    }
+
+    await CartArticle.create({ cartId, articleId });
+
+    const updatedCart = await Cart.findOne({
+      where: {
+        id: cartId,
+      },
+      include: [{ model: Article }],
+    });
+
```

Comment:
There is just to many database calls in this route that needs to be awaited for route to process.
The problem is that this bad design with cart handling can cause big performance issues when there is many users interacting with carts.
One problem you need to solve here is that in every route that interacts with cart you need to ask database to give you cart of the current user. So even if your system already did that once, next time user adds/removes/updates cart you getting same cart from database again and again.
Try to find solution how you dont need to fetch cart from database in every route.